app/src/main/java/org/thoughtcrime/securesms/home/HomeActivity.kt

```diff
@@ -773,9 +674,120 @@ class HomeActivity : PassphraseRequiredActionBarActivity(),
     // endregion
 }
 
-data class NameIdContact(val name: String?, val id: String, val contact: Contact): Comparable<NameIdContact> {
-    override fun compareTo(other: NameIdContact): Int = comparator.compare(this, other)
-    companion object {
-        val comparator = compareBy<NameIdContact>({ it.name }, { it.id })
+@Preview
+@Composable
+fun PreviewSeedReminder(
+    @PreviewParameter(SessionColorsParameterProvider::class) colors: Colors
+) {
+    PreviewTheme(colors) {
+        SeedReminder {}
+    }
+}
+
+@Composable
+private fun SeedReminder(startRecoveryPasswordActivity: () -> Unit) {
+    Column {
+        // Color Strip
+        Box(
+            Modifier
+                .fillMaxWidth()
+                .height(4.dp)
+                .background(LocalColors.current.primary)
+        )
+        Row(
+            Modifier
+                .background(LocalColors.current.backgroundSecondary)
+                .padding(
+                    horizontal = LocalDimensions.current.marginSmall,
+                    vertical = LocalDimensions.current.marginExtraSmall
+                )
+        ) {
+            Column(Modifier.weight(1f)) {
+                Row {
+                    Text(
+                        stringResource(R.string.save_your_recovery_password),
+                        style = h8
+                    )
+                    Spacer(Modifier.requiredWidth(LocalDimensions.current.itemSpacingExtraSmall))
+                    SessionShieldIcon()
+                }
+                Text(
+                    stringResource(R.string.save_your_recovery_password_to_make_sure_you_don_t_lose_access_to_your_account),
+                    style = small
+                )
+            }
+            Spacer(Modifier.width(LocalDimensions.current.marginExtraExtraSmall))
+            SlimOutlineButton(
+                text = stringResource(R.string.continue_2),
+                modifier = Modifier
+                    .align(Alignment.CenterVertically)
+                    .contentDescription(R.string.AccessibilityId_reveal_recovery_phrase_button),
+                onClick = { startRecoveryPasswordActivity() }
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+fun PreviewEmptyView(
+    @PreviewParameter(SessionColorsParameterProvider::class) colors: Colors
+) {
+    PreviewTheme(colors) {
+        EmptyView(newAccount = false)
+    }
+}
+
+@Preview
+@Composable
+fun PreviewEmptyViewNew(
+    @PreviewParameter(SessionColorsParameterProvider::class) colors: Colors
+) {
+    PreviewTheme(colors) {
+        EmptyView(newAccount = true)
+    }
+}
+
+@Composable
+private fun EmptyView(newAccount: Boolean) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .padding(horizontal = 50.dp)
+            .padding(bottom = 12.dp)
```

Comment:
Aren't all these paddings standardised?

Reply:
Color references aren't well specified in designs.
Standard dimension references are not specified at all.

Reply:
Yes but you have created a LocalDimensions object to unify them.
Don't you have multiple things that are using a horizontal padding of 50?
Isn't 12dp something reused also?

Reply:
@bemusementpark

Reply:
I had a hard time reading/making sense of these dimensions initially, but I can see now there's a standard margin for most of the content in onboarding and a standard margin for the continue button. done.

Reply:
oh wait, the bottom paddig is not supposed to be there, and the 50dp is bespoke. but I put it in Dimensions.